File: tl/src/main/java/com/github/badoualy/telegram/tl/api/TLUpdateEncryptedMessagesRead.java

package com.github.badoualy.telegram.tl.api;


import com.github.badoualy.telegram.tl.TLContext;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

import static com.github.badoualy.telegram.tl.StreamUtils.readInt;
import static com.github.badoualy.telegram.tl.StreamUtils.writeInt;



public class TLUpdateEncryptedMessagesRead extends TLAbsUpdate {
    public static final int CLASS_ID = 0x38fe25b7;

    public TLUpdateEncryptedMessagesRead() {

    }


    public TLUpdateEncryptedMessagesRead(        int _chatId,         int _maxDate,         int _date) {
        this.chatId = _chatId;
        this.maxDate = _maxDate;
        this.date = _date;

    }


    public int getClassId() {
        return CLASS_ID;
    }


    protected int chatId;

    protected int maxDate;

    protected int date;


    public int getChatId() {
        return chatId;
    }

    public void setChatId(int value) {
        this.chatId = value;
    }

    public int getMaxDate() {
        return maxDate;
    }

    public void setMaxDate(int value) {
        this.maxDate = value;
    }

    public int getDate() {
        return date;
    }

    public void setDate(int value) {
        this.date = value;
    }


    @Override
    public void serializeBody(OutputStream stream) throws IOException {

        writeInt(this.chatId, stream);
        writeInt(this.maxDate, stream);
        writeInt(this.date, stream);
    }


    @Override
    public void deserializeBody(InputStream stream, TLContext context) throws IOException {

        this.chatId = readInt(stream);
        this.maxDate = readInt(stream);
        this.date = readInt(stream);
    }



    @Override
    public String toString() {
        return "updateEncryptedMessagesRead#38fe25b7";
    }

}

File: tl/src/main/kotlin/com/github/badoualy/telegram/tl/api/TLUpdateEncryptedMessagesRead.kt
package com.github.badoualy.telegram.tl.api

import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_BOOLEAN
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_CONSTRUCTOR_ID
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_DOUBLE
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_INT32
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_INT64
import com.github.badoualy.telegram.tl.TLObjectUtils.computeTLBytesSerializedSize
import com.github.badoualy.telegram.tl.TLObjectUtils.computeTLStringSerializedSize
import com.github.badoualy.telegram.tl.serialization.TLDeserializer
import com.github.badoualy.telegram.tl.serialization.TLSerializer
import java.io.IOException
import kotlin.Any
import kotlin.Boolean
import kotlin.Int
import kotlin.String
import kotlin.jvm.Throws

/**
 * updateEncryptedMessagesRead#38fe25b7
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
class TLUpdateEncryptedMessagesRead() : TLAbsUpdate() {
    var chatId: Int = 0

    var maxDate: Int = 0

    var date: Int = 0

    private val _constructor: String = "updateEncryptedMessagesRead#38fe25b7"

    override val constructorId: Int = CONSTRUCTOR_ID

    constructor(
            chatId: Int,
            maxDate: Int,
            date: Int
    ) : this() {
        this.chatId = chatId
        this.maxDate = maxDate
        this.date = date
    }

    @Throws(IOException::class)
    override fun serializeBody(tlSerializer: TLSerializer) = with (tlSerializer)  {
        writeInt(chatId)
        writeInt(maxDate)
        writeInt(date)
    }

    @Throws(IOException::class)
    override fun deserializeBody(tlDeserializer: TLDeserializer) = with (tlDeserializer)  {
        chatId = readInt()
        maxDate = readInt()
        date = readInt()
    }

    override fun computeSerializedSize(): Int {
        var size = SIZE_CONSTRUCTOR_ID
        size += SIZE_INT32
        size += SIZE_INT32
        size += SIZE_INT32
        return size
    }

    override fun toString() = _constructor

    override fun equals(other: Any?): Boolean {
        if (other !is TLUpdateEncryptedMessagesRead) return false
        if (other === this) return true

        return chatId == other.chatId
                && maxDate == other.maxDate
                && date == other.date
    }
    companion object  {
        const val CONSTRUCTOR_ID: Int = 0x38fe25b7.toInt()
    }
}
